Modules/IO/GDCM/src/itkGDCMImageIO.cxx

```diff
@@ -1196,12 +1255,31 @@ GDCMImageIO::Write(const void * buffer)
     pixeldata.SetByteValue(copyBuffer, static_cast<uint32_t>(len));
     delete[] copyBuffer;
   }
+  else if (this->GetNumberOfComponents() == 3 && m_OutputRGBtoYBR)
+  {
+    // Convert RGB to Y'CbCr before RLE compression
+    if (m_CompressionType == TCompressionType::LOSSYJPEG)
+    {
+      itkWarningMacro(<< "OutputRGBtoYBR is ignored for LOSSYJPEG");
+      pixeldata.SetByteValue(static_cast<const char *>(buffer), static_cast<unsigned int>(numberOfBytes));
+    }
+    else
+    {
+      if (m_CompressionType != TCompressionType::RLE)
+      {
+        itkWarningMacro(<< "OutputRGBtoYBR will be applied, but is not supported other than with RLE compression");
+      }
+      auto * copy = new char[len];
```

Comment:
Please use a modern C++ container such as `std::vector` which uses RAII so we do not have to be burdened with considerations for exceptions and memory leaks.

Reply:
It is re-used old code, if you drop a line, i'll replace, not sure how exactly do you like. Thank you.